README.md

```diff
@@ -0,0 +1,143 @@
+main.py
+```  
+
+The program should now be running smoothly on the computer. 
+
+## Microprocessor Specifications 
+
+The microprocessor that is being simulated has a 4 KB memory. The instructions are always stored in even-numbered memory addresses, while the data can be stored anywhere. Instructions occupy 16 bits, while other data occupies 8. The microprocessor has eight 8-bit registers, from R0 to R7. R0 is always zero, and R1 will serve as accumulator for certain instructions. In addition to these registers, the microprocessor also counts with an 11-bit Program Counter, a 12-bit Stack Pointer, and a 16-bit Instruction Register. 
+
+## I/O Modules 
+
+The microprocessor simulator counts with several I/O modules in its interface. Each one displays the content of a specified memory address, with the exception of one module that changes the content of the specified port. The memory locations utilized as ports can be changed to any available address. Below is I/O Module explained.  
+
+### Traffic Light 
+
+The traffic light module consists of two traffic light widgets. This feature reads the content found on the port specified by the user. The first six bits will represent the six lights on the two traffic lights in top-to-bottom order, with the left traffic light representing the first three bits and the right traffic light showing the other three. A one on the bit would represent an activated light, whereas a zero value would show a turned off light. For example, the six bits "100100" would show the two red lights of each traffic light turned on. The final two bits decide whether or not the content shown by the activated traffic lights should be intermittent. If the two least significant bits are both 1, then the lights will be blinking. To use the past example, the bits "10010011" would show the two red lights blinking. 
+
+### Seven Segment Display 
+
+The 7-segment display shows two digits with a seven-segment configuration. This feature utilizes the content of one 8-bit memory address specified by the user. The 7 most significant bits represent the seven segments of a digit: a,b,c,d,e,f & g, respectively. For example, the seven bits "1000101" would show the segments a, e & g. The remaining, least significant bit decides which one of the two digits will display the segments activated by the bits: A zero would show the seven segments on the left digit, while a one would show the seven segments on the right digit. Thus, the byte "10001011" would show the segments a, e & g turned on for the right-side digit.  
+
+### ASCII Grid 
+
+The ASCII Grid is the only I/O module that needs more than one memory address. The ASCII Grid consists of eight squares, each one representing one memory address. Each square on the grid will display the ASCII character that is represented by the eight bits shown in that specific memory address. For example, the memory content "10000001" would show the letter 'A' in the square corresponding to that memory address.
```

Comment:
The ASCII Grid is the only I/O module that needs more than one memory address. Eight contiguous memory addresses are reserved for this I/O.